### PR TITLE
Cash 414 is appeal banner shrunk

### PR DESCRIPTION
**Pull request by mcstover**

### CODE REVIEW

**Comment by mcstover** [comment]
@BoulderBrains @ksekhri This update moves a lot around but ultimately gets all of the setting and session checks in place and uses them to show an hide the various content blocks. It will probably be easiest to get this merged into 411 before commencing any other work.

We also hide the appeal on checkout and save the users action of closing or opening the appeal banner back to their site session ( this is dependent upon https://github.com/kiva/kiva/pull/6844 but should just console the error until it's present )

File: src/components/WwwFrame/EndOfYearAppealBanner/AppealBanner.vue [hunk 1]
```diff
@@ -1,22 +1,17 @@
 <template>
-	<div class="appeal-banner-wrapper">
+	<div class="appeal-banner-wrapper" v-if="showAppeal">
```
Comment (mcstover) [inline src/components/WwwFrame/EndOfYearAppealBanner/AppealBanner.vue]: @BoulderBrains We had lost the visibility check somewhere along the way. This new variable ensures at least 1 of the Appeal features is turned on and doesn't show if we're on checkout.

File: src/components/WwwFrame/EndOfYearAppealBanner/AppealBanner.vue [hunk 2]
```diff
@@ -1,22 +1,17 @@
...
+					<h2>
 						<!-- IF BONUS APPEAL BANNER -->
-						<!-- PUT CONDITIONAL HERE -->
-						<!-- <h2>Donate today and receive a bonus to lend!</h2> -->
-
+						<span v-if="appealBonusEnabled">Donate today and receive a bonus to lend!</span>
+						<!-- IF MATCHED APPEAL BANNER -->
+						<span v-else-if="appealMatchEnabled">Double or triple the impact of your donation!</span>
+						<!-- ELSE STANDARD APPEAL BANNER -->
+						<span v-else>Your donation will power impact and hope in 2019</span>
```
Comment (mcstover) [inline src/components/WwwFrame/EndOfYearAppealBanner/AppealBanner.vue]: It's easiest to put our Default content as the last option in the v-if cascade.

File: src/graphql/query/appealBanner.graphql [hunk 3]
```diff
@@ -1,10 +1,29 @@
 query {
 	general {
-		setting(setting: "feature.endofyear_appeal.enabled") {
+		appeal_enabled: setting(setting: "feature.endofyear_appeal.enabled") {
```
Comment (mcstover) [inline src/graphql/query/appealBanner.graphql]: Aliases for all settings and checks against their start and end time.

File: src/components/WwwFrame/EndOfYearAppealBanner/AppealBanner.vue [hunk 4]
```diff
@@ -101,19 +93,68 @@ export default {
...
 		query: appealBannerQuery,
 		preFetch: true,
 		result({ data }) {
-			this.isVisible = _get(data, 'general.setting.value') === 'true' || false;
+			this.appealEnabled = settingEnabled(
+				data,
+				'general.appeal_enabled.value',
+				'general.appeal_start_time.value',
+				'general.appeal_end_time.value'
+			);
```
Comment (mcstover) [inline src/components/WwwFrame/EndOfYearAppealBanner/AppealBanner.vue]: Check enabled + start and end times.

File: src/components/WwwFrame/EndOfYearAppealBanner/AppealBanner.vue [hunk 5]
```diff
@@ -101,19 +93,68 @@ export default {
...
+				'general.appeal_match_enabled.value',
+				'general.appeal_match_start_time.value',
+				'general.appeal_match_end_time.value'
+			);
+
+			// This setting SHOULD be temporary and CANNOT reveal this appeal alone.
+			this.appealBonusEnabled = readBoolSetting(data, 'general.appeal_bonus_active.value');
+
+			// the IsAppealBannerShrunk session value returns either false or '"1"'
+			this.isAppealShrunk = readJSONSetting(data, 'general.appeal_banner_shrunk.data') === 1;
```
Comment (mcstover) [inline src/components/WwwFrame/EndOfYearAppealBanner/AppealBanner.vue]: Check session for appeal shrunk, then watch this value below to close the appeal on load if this is set to true.

File: src/components/WwwFrame/EndOfYearAppealBanner/AppealBanner.vue (hunk 1 above)
Comment (BoulderBrains) [reply src/components/WwwFrame/EndOfYearAppealBanner/AppealBanner.vue]: 🆒

File: src/components/WwwFrame/EndOfYearAppealBanner/AppealBanner.vue (hunk 2 above)
Comment (BoulderBrains) [reply src/components/WwwFrame/EndOfYearAppealBanner/AppealBanner.vue]: Makes sense 👍

File: src/components/WwwFrame/EndOfYearAppealBanner/AppealBanner.vue (hunk 4 above)
Comment (BoulderBrains) [reply src/components/WwwFrame/EndOfYearAppealBanner/AppealBanner.vue]: This makes more sense than what was in there. 🧠Looking at all the setting options.

File: src/components/WwwFrame/EndOfYearAppealBanner/AppealBanner.vue [hunk 6]
```diff
@@ -101,19 +93,68 @@ export default {
...
 		},
 	},
+	computed: {
+		showAppeal() {
+			// make sure the appeal is enable + we're not on checkout
+			return (this.appealEnabled || this.appealMatchEnabled) && this.$route.path !== '/checkout';
```
Comment (BoulderBrains) [inline src/components/WwwFrame/EndOfYearAppealBanner/AppealBanner.vue]: Nice or/and statement here to hide it on checkout.

File: src/graphql/query/appealBanner.graphql (hunk 3 above)
Comment (BoulderBrains) [reply src/graphql/query/appealBanner.graphql]: Nice, it's great we can pass through and check the group vs all the individual settings.

**Pull request merged**